improver/wxcode/weather_symbols.py

```diff
@@ -479,28 +481,30 @@ def find_all_routes(graph, start, end, omit_nodes=None, route=None):
         return routes
 
     @staticmethod
-    def create_symbol_cube(cube):
+    def create_symbol_cube(cubes):
         """
-        Create an empty weather_symbol cube initialised with -1 across the
-        grid.
+        Create an empty weather_symbol cube
 
         Args:
-            cube (iris.cube.Cube):
-                An x-y slice of one of the input cubes, used to define the
-                size of the weather symbol grid.
+            cubes (list or iris.cube.CubeList):
+                List of input cubes used to generate weather symbols
         Returns:
             iris.cube.Cube:
-                A cube full of -1 values, with suitable metadata to describe
-                the weather symbols that will fill it.
+                A cube with suitable metadata to describe the weather symbols
+                that will fill it
         """
-        threshold_coord = find_threshold_coordinate(cube)
-        cube_format = next(cube.slices_over([threshold_coord]))
-        symbols = cube_format.copy(data=np.full(cube_format.data.shape, -1,
-                                                dtype=np.int32))
-
-        symbols.remove_coord(threshold_coord)
-        symbols = add_wxcode_metadata(symbols)
-
+        threshold_coord = find_threshold_coordinate(cubes[0])
+        template_cube = next(cubes[0].slices_over([threshold_coord])).copy()
+        # remove coordinates and bounds that do not apply to weather symbols
+        template_cube.remove_coord(threshold_coord)
+        for coord in template_cube.coords():
+            if coord.name() in ['forecast_period', 'time']:
+                coord.bounds = None
+
+        attributes = generate_mandatory_attributes(cubes)
+        symbols = create_new_diagnostic_cube(
```

Comment:
I just noticed that `create_new_diagnostic_cube` doesn't have a unit test for the exception that is raised within `create_new_diagnostic_cube`. Do you think that it would be useful to add a unit test (not necessarily part of this PR)?

Reply:
Good spot.  I've added a unit test for this.